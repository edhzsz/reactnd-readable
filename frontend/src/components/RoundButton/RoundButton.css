.btn-circle {
  display: block;
  border-radius: 50%;
  box-shadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
  font-size: 0;
  background-repeat: no-repeat;
  background-position: center;
}