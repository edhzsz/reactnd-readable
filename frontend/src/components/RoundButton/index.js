import React from 'react';
import PropTypes from 'prop-types';
import './RoundButton.css'

const RoundButton = (props) => {
  return (
    <div className={props.className}>
      <a className={`btn-circle ${ props.type }`} href="#addpost">{props.children}</a>
    </div>
  );
}

RoundButton.propTypes = {
  className: PropTypes.string,
  type: PropTypes.oneOf(['primary', 'secondary']).isRequired,
}

// Specifies the default values for props:
RoundButton.defaultProps = {
  type: 'primary'
};

export default RoundButton;