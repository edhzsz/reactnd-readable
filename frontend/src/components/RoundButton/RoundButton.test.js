import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RoundButton from '.';

const wrapper = shallow(<RoundButton/>);

describe('Component: RoundButton', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<RoundButton />, div);
  });

  it('should match its empty snapshot', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });

  it('should match its empty snapshot', () => {
    const button = shallow(<RoundButton className="add-post primary">Add a Book</RoundButton>)
    const tree = toJson(button);
    expect(tree).toMatchSnapshot();
  });

});