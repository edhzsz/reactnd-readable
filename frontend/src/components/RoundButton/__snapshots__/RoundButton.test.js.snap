// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: RoundButton should match its empty snapshot 1`] = `
<div>
  <a
    className="btn-circle primary"
    href="#addpost"
  />
</div>
`;

exports[`Component: RoundButton should match its empty snapshot 2`] = `
<div
  className="add-post primary"
>
  <a
    className="btn-circle primary"
    href="#addpost"
  >
    Add a Book
  </a>
</div>
`;
